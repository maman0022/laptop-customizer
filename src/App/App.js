import React, { Component } from 'react';
import store from '../STORE';
import Nav from '../Nav/Nav';
import MainForm from '../MainForm/MainForm';
import MainSummary from '../MainSummary/MainSummary';
import './App.css';

// This object will allow us to
// easily convert numbers into US dollar values
export const USCurrencyFormat = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD'
});

class App extends Component {
  state = {
    selected: {
      Processor: {
        name: '17th Generation Intel Core HB (7 Core with donut spare)',
        cost: 700
      },
      'Operating System': {
        name: 'Ubuntu Linux 16.04',
        cost: 200
      },
      'Video Card': {
        name: 'Toyota Corolla 1.5v',
        cost: 1150.98
      },
      Display: {
        name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
        cost: 1500
      }
    }
  };

  updateFeature = (feature, newValue) => {
    const selected = Object.assign({}, this.state.selected);
    selected[feature] = newValue;
    this.setState({
      selected
    });
  };

  render() {
    const total = Object.keys(this.state.selected).reduce(
      (acc, curr) => acc + this.state.selected[curr].cost,
      0
    );

    return (
      <div className="App">
        <Nav />
        <main>
          <MainForm USCurrencyFormat={USCurrencyFormat} state={this.state} updateFeature={this.updateFeature} store={store}/>
          <MainSummary USCurrencyFormat={USCurrencyFormat} total={total} state={this.state} />
        </main>
      </div>
    );
  }
}

export default App;