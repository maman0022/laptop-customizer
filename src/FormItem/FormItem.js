import React from 'react';
import slugify from 'slugify';
import './FormItem.css'

export default function FormItem(props) {
  return Object.keys(props.store.FEATURES).map((feature, idx) => {
    const featureHash = feature + '-' + idx;
    const options = props.store.FEATURES[feature].map(item => {
      const itemHash = slugify(JSON.stringify(item));
      return (
        <div key={itemHash} className="feature__item">
          <input
            type="radio"
            id={itemHash}
            className="feature__option"
            name={slugify(feature)}
            checked={item.name === props.state.selected[feature].name}
            onChange={e => props.updateFeature(feature, item)}
          />
          <label htmlFor={itemHash} className="feature__label">
            {item.name} ({props.USCurrencyFormat.format(item.cost)})
          </label>
        </div>
      );
    });

    return (
      <fieldset className="feature" key={featureHash}>
        <legend className="feature__name">
          <h3>{feature}</h3>
        </legend>
        {options}
      </fieldset>
    );
  });
}