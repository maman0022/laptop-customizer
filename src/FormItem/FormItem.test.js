import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormItem from './FormItem';
import { USCurrencyFormat } from '../App/App';

let state = {
  selected: {
    Processor: {
      name: '17th Generation Intel Core HB (7 Core with donut spare)',
      cost: 700
    },
    'Operating System': {
      name: 'Ubuntu Linux 16.04',
      cost: 200
    },
    'Video Card': {
      name: 'Toyota Corolla 1.5v',
      cost: 1150.98
    },
    Display: {
      name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
      cost: 1500
    }
  }
};

let updateFeature = null;

let store = {
  FEATURES: {
    Processor: [
      {
        name: '17th Generation Intel Core HB (7 Core with donut spare)',
        cost: 700
      }
    ],
    "Operating System": [
      {
        name: 'Ubuntu Linux 16.04',
        cost: 200
      }
    ],
    "Video Card": [
      {
        name: 'Toyota Corolla 1.5v',
        cost: 1150.98
      }
    ],
    Display: [
      {
        name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
        cost: 1500
      }
    ]
  }
}

describe('FormItem Component', () => {
  it('an array of form items renders with passed in dummy data', () => {
    const wrapper = shallow(<FormItem USCurrencyFormat={USCurrencyFormat} state={state} updateFeature={updateFeature} store={store} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})