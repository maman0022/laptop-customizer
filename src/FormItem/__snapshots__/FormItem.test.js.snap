// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItem Component an array of form items renders with passed in dummy data 1`] = `
Array [
  <fieldset
    className="feature"
    key="Processor-0"
  >
    <legend
      className="feature__name"
    >
      <h3>
        Processor
      </h3>
    </legend>
    <div
      className="feature__item"
      key="\\"name\\":\\"17th-Generation-Intel-Core-HB-(7-Core-with-donut-spare)\\"\\"cost\\":700"
    >
      <input
        checked={true}
        className="feature__option"
        id="\\"name\\":\\"17th-Generation-Intel-Core-HB-(7-Core-with-donut-spare)\\"\\"cost\\":700"
        name="Processor"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"17th-Generation-Intel-Core-HB-(7-Core-with-donut-spare)\\"\\"cost\\":700"
      >
        17th Generation Intel Core HB (7 Core with donut spare)
         (
        $700.00
        )
      </label>
    </div>
  </fieldset>,
  <fieldset
    className="feature"
    key="Operating System-1"
  >
    <legend
      className="feature__name"
    >
      <h3>
        Operating System
      </h3>
    </legend>
    <div
      className="feature__item"
      key="\\"name\\":\\"Ubuntu-Linux-16.04\\"\\"cost\\":200"
    >
      <input
        checked={true}
        className="feature__option"
        id="\\"name\\":\\"Ubuntu-Linux-16.04\\"\\"cost\\":200"
        name="Operating-System"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"Ubuntu-Linux-16.04\\"\\"cost\\":200"
      >
        Ubuntu Linux 16.04
         (
        $200.00
        )
      </label>
    </div>
  </fieldset>,
  <fieldset
    className="feature"
    key="Video Card-2"
  >
    <legend
      className="feature__name"
    >
      <h3>
        Video Card
      </h3>
    </legend>
    <div
      className="feature__item"
      key="\\"name\\":\\"Toyota-Corolla-1.5v\\"\\"cost\\":1150.98"
    >
      <input
        checked={true}
        className="feature__option"
        id="\\"name\\":\\"Toyota-Corolla-1.5v\\"\\"cost\\":1150.98"
        name="Video-Card"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"Toyota-Corolla-1.5v\\"\\"cost\\":1150.98"
      >
        Toyota Corolla 1.5v
         (
        $1,150.98
        )
      </label>
    </div>
  </fieldset>,
  <fieldset
    className="feature"
    key="Display-3"
  >
    <legend
      className="feature__name"
    >
      <h3>
        Display
      </h3>
    </legend>
    <div
      className="feature__item"
      key="\\"name\\":\\"15.6\\"-UHD-(3840-x-2160)-60Hz-Bright-Lights-and-Knobs\\"\\"cost\\":1500"
    >
      <input
        checked={true}
        className="feature__option"
        id="\\"name\\":\\"15.6\\"-UHD-(3840-x-2160)-60Hz-Bright-Lights-and-Knobs\\"\\"cost\\":1500"
        name="Display"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"15.6\\"-UHD-(3840-x-2160)-60Hz-Bright-Lights-and-Knobs\\"\\"cost\\":1500"
      >
        15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs
         (
        $1,500.00
        )
      </label>
    </div>
  </fieldset>,
]
`;
